### PR TITLE
Issue32

### PR DESCRIPTION
**Pull request by mattj23**
Pull request for Issue #32 submitted for review.

This one turned into something big.  Changed default tolerances to 1e-10, since 1e-6 (which was on the Vector3D class's IsParallelTo method) ended up producing incorrect results with vectors different in the third decimal place.

Lots of unit tests, should be 100% coverage of the methods I added and the old IsParallelTo methods.  

I did not remove all of the old tests, but maybe they ought to be.

### CODE REVIEW

**Comment by JohanLarsson** [comment]
Thanks!

File: src/Spatial/Euclidean/Line2D.cs [hunk 1]
```diff
@@ -129,12 +135,53 @@ public Point2D ClosestPointTo(Point2D p, bool mustBeOnSegment)
...
 
+        /// <summary>
+        /// Checks to determine whether or not two lines are parallel to each other, using the dot product within 
+        /// the double precision specified in the MathNet.Numerics package.
+        /// </summary>
+        /// <param name="other">The other line to check this one against</param>
+        /// <returns>True if the lines are parallel, false if they are not</returns>
+        public bool IsParallelTo(Line2D other)
+        {
+            return this.Direction.IsParallelTo(other.Direction, Precision.DoublePrecision * 2);
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/Line2D.cs]: What was the reason for the *2 again?

Comment (mattj23) [inline src/Spatial/Euclidean/Line2D.cs]: The test case failed with two identical vectors when it was just Precision.DoublePrecision.  In fact, the mismatch was exactly DoublePrecision * 2, so the comparison had to become a <= in order for it to return correct results.

File: src/SpatialUnitTests/Euclidean/Line2DTests.cs [hunk 2]
```diff
@@ -174,8 +175,35 @@ public void IntersectWithTest(string s1, string e1, string s2, string e2, string
...
+        [TestCase("0,0", "0.00001,-1.0000", "1,1", "1,2", false)]
+        public void IsParallelToWithinDoubleTol(string s1, string e1, string s2, string e2, bool expected)
+        {
+            var line1 = Line2D.Parse(s1, e1);
+            var line2 = Line2D.Parse(s2, e2);
+            
+            Assert.AreEqual(expected, line1.IsParallelTo(line2));
+        }
+
+        [TestCase("0,0", "0,1", "1,1", "1,2", 0.01, true)]
```
Comment (JohanLarsson) [inline src/SpatialUnitTests/Euclidean/Line2DTests.cs]: `[TestCase("0,0", "0,1", "0,1", "0,0", 0.01, true|false)]`
Here we a have a decision to make.

Comment (mattj23) [inline src/SpatialUnitTests/Euclidean/Line2DTests.cs]: A friend of mine is a professor of mathematics, I'll reach out to him and see what he thinks.  If he can't answer the question I suspect he'll know someone who can.

**Comment by JohanLarsson** [comment]
@mattj23 sorry about the delay, really busy here.

**Pull request merged**